lnbits/cache.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+import asyncio
+from time import time
+from typing import Any, NamedTuple, Optional
+
+from loguru import logger
+
+
+class Cached(NamedTuple):
+    value: Any
+    expiry: float
+
+
+def _add_prefix(key: str, prefix: str):
+    return prefix + ":" + key if prefix else key
+
+
+class Cache:
+    """
+    Small caching utility providing simple get/set interface (very much like redis)
+    """
+
+    def __init__(self):
+        self._values: dict[Any, Cached] = {}
+
+    def get(self, key: str, prefix: str = "") -> Optional[Any]:
+        cached = self._values.get(_add_prefix(key, prefix))
+        if cached is not None:
+            if cached.expiry > time():
+                return cached.value
+            else:
+                self.pop(key, prefix)
+        return None
+
+    def set(self, key: str, value: Any, expiry: float = 10, prefix: str = ""):
+        self._values[_add_prefix(key, prefix)] = Cached(value, time() + expiry)
+
+    def pop(self, key: str, prefix: str = "", default=None) -> Optional[Any]:
+        cached = self._values.pop(_add_prefix(key, prefix), None)
+        if cached:
+            return cached.value
+        return default
+
+    async def save_result(self, coro, key: str, expiry: float = 10, prefix: str = ""):
```

Comment:
- order of params should be consistent with the other methods
- what is the advantage of this functionality? 
   - `coro` can be called at the line before `save_results` and the results passed to `save_results`
   - it only saves one line of code

Reply:
- if `coro` has params then a wrapper haws to be created
- I would remove this function, keep the interface as simple as possible

Reply:
> * it only saves one line of code

```python
rate = cache.get("ccy-usd")
if not rate:
    rate = await fetch_rate("usd")
    cache.set("ccy-usd", rate)
```
vs
```python
rate = await cache.save_result(lambda: fetch_rate("usd") key="ccy-usd")
```